### DISCUSSION TITLE
How to Disable Log and Metric Providers to Stop Instrumentation for Logs and Metrics?

### DISCUSSION THREAD

**Discussion by shahparthiv** [open-telemetry/opentelemetry-java-instrumentation#12561] [Q&A]
Hi everyone,

I'm working on a project where we need to temporarily disable the logging and metrics instrumentation to reduce resource usage. I've looked through the documentation but haven't found a clear way to fully disable log and metric providers. 

Is there a configuration setting, environment variable, or other approach to entirely disable these providers? Any advice or references would be greatly appreciated!

Thanks in advance!

**Comment by laurit** [comment]
Try setting `OTEL_METRICS_EXPORTER=none` and `OTEL_LOGS_EXPORTER=none`.

**Reply by h0cheung** [reply]
With these configs, I still found something like `io.opentelemetry.javaagent.instrumentation.log4j.appender.v2_17.Log4jHelper.capture(Logger, String, StackTraceElement, Level, Marker, Message, Throwable)` are costing CPU.

**Reply by laurit** [reply]
This isn't unexpected. Setting the exporter to `none` doesn't disable the instrumentation. When this discussion was created we didn't have a good way to detect whether the exporter was disabled or not. Now this might be possible using https://github.com/open-telemetry/opentelemetry-java/blob/9d4f61ad0b0abfe90594d73d11bd328fead0dfca/api/incubator/src/main/java/io/opentelemetry/api/incubator/logs/ExtendedLogger.java#L20 You should create a new issue if you wish for someone to look at this. It is unlikely that anybody will work on this based on a comment in an old discussion. In your issue it would be nice if you could elaborate whether the cpu consumption of `Log4jHelper.capture` is significant or not. If it is significant then it would be nice to know what exactly there consumes the cpu.
